File: python_sdk/src/reality_capture/service/bucket.py
from pydantic import BaseModel, Field
from reality_capture.service.reality_data import ContainerLinks


class Bucket(BaseModel):
    itwin_id: str = Field(description="iTwin Id for the bucket.", alias="iTwinId")


class BucketResponse(BaseModel):
    bucket: Bucket = Field(description="Bucket information")
    links: ContainerLinks = Field(description="The link to the container.", alias="_links")

File: typescript_sdk/src/reality_capture/service/bucket.ts
import { z } from "zod";
import { ContainerLinks } from "./reality_data";

export const Bucket = z.object({
  iTwinId: z.string().describe("iTwin Id for the bucket.")
});
export type Bucket = z.infer<typeof Bucket>;

export const BucketResponse = z.object({
  bucket: Bucket.describe("Bucket information"),
  _links: ContainerLinks.describe("The link to the container.")
});
export type BucketResponse = z.infer<typeof BucketResponse>;